src/Resources/app/administration/src/module/sw-extension-store/component/sw-extension-buy-modal/index.js

```diff
@@ -255,6 +286,107 @@ export default {
             this.$emit('modal-close');
         },
 
+        getDiscountClasses(variant) {
+            return {
+                'is--discounted': this.hasActiveDiscount(variant),
+            };
+        },
+
+        getVariantPrice(variant) {
+            return this.extensionStoreService.getCalculatedPrice(variant);
+        },
```

Comment:
Note that the service is also available in the template, there is no need to proxy all methods